### ISSUE TITLE
Support System.Text.Json JsonStringEnumMemberNameAttribute

### ISSUE THREAD

**Issue by bdovaz** [RicoSuter/NJsonSchema#1738]
Added in 9.0.0: https://github.com/dotnet/runtime/issues/74385

**Comment by desjoerd** [comment]
Currently we do this in the file `Enum.Member.Annotations.liquid`. Maybe it's helpfull for someone :).

```liquid
{%  if IsStringEnum %}
    [System.Text.Json.Serialization.JsonStringEnumMemberName(@"{{ enum.Value | replace: '"', '""' }}")]
{% endif %}
```

**Comment by lh-ak** [comment]
It seems that this is now implemented in the newest version, but as @bdovaz already mentioned it's available in > .NET 9. for .NET 8 it brakes the client generation currently as it's not available there:

<img width="1861" height="62" alt="Image" src="https://github.com/user-attachments/assets/9c724221-8fae-4ed8-a267-a32c5d50066a" />

Is it possible to do this **only** for .NET versions above 8?

**Comment by RicoSuter** [comment]
/cc @lahma @0xced

We need to fix this ASAP... major breaking change for non-latest .NET users. I think we need to introduce CSharpVersion configuration property (same as TypeScriptVersion property).

**Comment by desjoerd** [comment]
Or emit a `#if NET9_0_OR_GREATER` in the template

**Comment by RicoSuter** [comment]
> Or emit a `#if NET9_0_OR_GREATER` in the template

yeah good idea, simple and does not need yet another config :-)

**Comment by 0xced** [comment]
There's no need for .NET 9, only the [System.Text.Json version 9](https://www.nuget.org/packages/System.Text.Json/9.0.0) package. It targets .NET Standard 2.0 so I don't see it as a big deal, is it?

Also, yes, the new `JsonStringEnumMemberName` attribute is needed to properly serialize an enum value that does not match the enum name exactly. The `System.Runtime.Serialization.EnumMember` has never been supported by System.Text.Json, there was a long thread at https://github.com/dotnet/runtime/issues/74385.

**Comment by 0xced** [comment]
> So it's either a config option or requiring >= NET9 for this feature.

No, it doesn't **require** .NET 9. You can just add a package reference to [System.Text.Json](https://www.nuget.org/packages/System.Text.Json/9.0.9) (version 9 onwards) and it will work just fine on any plaform supporting .NET Standard 2.0, which includes .NET 8. It will even work on .NET Framework 4.6.2.

I agree that it's a bit counterintuitive but you can get `System.Text.Json` features introduced in .NET 9 with just a package reference without needing to change your project target framework.

**Comment by desjoerd** [comment]
I know, but adding a config option for one year when System.Text.Json 8.x goes out of support is a lot. So add the check for NET9_0_OR_GREATER and when Net8 is out of support you can remove this and make it available for NET Framework.
